models/marts/core/marketing/dim_customers.sql

```diff
@@ -0,0 +1,106 @@
+with customers as (
+    select * from {{ ref('stg_ecommerce__customers') }}
+),
+
+orders as (
+    select * from {{ ref('stg_ecommerce__orders') }}
+),
+
+order_items as (
+    select * from {{ ref('stg_ecommerce__order_items') }}
+),
+
+order_reviews as (
+    select * from {{ ref('stg_ecommerce__order_reviews') }}
+),
+
+cancelled_missing_orders as (
+    select distinct order_id
+    from orders
+    where order_status not in ("canceled", "unavailable")
+),
+
+first_n_last_order as (
+    select
+        customers.customer_id,
+        min(orders.order_approved_at) as first_order_approved_at,
+        max(orders.order_approved_at) as last_order_approved_at,
+        date_diff(max(orders.order_approved_at), min(orders.order_approved_at), day) as customer_lifespan
+
+    from customers
+    left outer join orders
+        on customers.customer_id = orders.customer_id
+    inner join cancelled_missing_orders -- Inner Join with cancelled_missing_orders ensures that Orders that are either cancelled or unavailable are not factored in when determining a customer's lifespan
```

Comment:
why do this with a join? Why not just apply that filter and then just select from a filtered table rather than joining to orders and then inner joining to remove cancelled and missing orders?

Reply:
@JFrackson  Just to be sure, you're suggesting applying the filter within orders received from staging (after Line 6) , right ? 
![image](https://user-images.githubusercontent.com/89142371/186924154-bf84bd1e-8eec-4887-9e50-63e2e0a35f93.png)